### PR TITLE
Upgrading Apache Helix to 1.3.1 version

### PR DESCRIPTION
**Pull request by soumitra-st**
This PR is a followup to https://github.com/apache/pinot/pull/11450, which was reverted because of an Helix issue.

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Please also upgrade the license, see #11560 for reference

File: LICENSE-binary [hunk 1]
```diff
@@ -404,8 +404,7 @@ org.apache.flink:flink-shaded-netty:4.1.49.Final-12.0
 org.apache.flink:flink-shaded-zookeeper-3:3.4.14-12.0
 org.apache.flink:flink-streaming-java_2.12:1.12.0
 org.apache.flink:force-shading:1.12.0
-org.apache.helix:helix-common:1.0.4
```
Comment (soumitra-st) [inline LICENSE-binary:407]: helix-common is not used, so removed it.

Comment (Jackie-Jiang) [reply LICENSE-binary:407]: Are you sure? helix-common is pulled by helix-core. You are missing the following 3 lines as well

Comment (soumitra-st) [reply LICENSE-binary:407]: I thought only direct dependencies are listed in the LICENSE file. Brought back the helix-common and fixed license in the rest of helix libs.

File: LICENSE-binary [hunk 2]
```diff
@@ -404,11 +404,11 @@ org.apache.flink:flink-shaded-netty:4.1.49.Final-12.0
...
-org.apache.helix:helix-core:1.0.4
-org.apache.helix:metadata-store-directory-common:1.0.4
-org.apache.helix:metrics-common:1.0.4
-org.apache.helix:zookeeper-api:1.0.4
+org.apache.helix:helix-core:1.3.1
```
Comment (Jackie-Jiang) [inline LICENSE-binary:407]:
```suggestion
org.apache.helix:helix-common:1.3.1
```

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**